api/src/devcomp/domain/services/solution-service-qrocm-ind.js

```diff
@@ -37,40 +37,56 @@ function match({ answerValue, solution }) {
 }
 
 function _applyTreatmentsToSolutions(solutions, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(solutions, (solution, solutionKey) => {
-    solution.forEach((variant, variantIndex) => {
-      if (qrocBlocksTypes[solutionKey] === 'select') {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, []);
-      } else {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, enabledTreatments);
+  const treatedSolutions = {};
+  for (const solutionKey in solutions) {
+    const solutionVariants = solutions[solutionKey];
+    const solutionType = qrocBlocksTypes[solutionKey];
+
+    treatedSolutions[solutionKey] = solutionVariants.map((variant) => {
+      switch (solutionType) {
+        case 'select':
+          return applyTreatments(variant, []);
+        case 'text':
+        default:
+          return applyTreatments(variant, enabledTreatments);
       }
     });
-  });
+  }
+
+  return treatedSolutions;
 }
 
 function _applyTreatmentsToAnswers(answers, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(answers, (answer, answerKey) => {
-    if (qrocBlocksTypes[answerKey] === 'select') {
-      answers[answerKey] = applyTreatments(answer, []);
-    } else {
-      answers[answerKey] = applyTreatments(answer, enabledTreatments);
+  const treatedAnswers = {};
+  for (const answerKey in answers) {
+    const answer = answers[answerKey];
```

Comment:
A chaque fois on fait ces deux lignes, qui correspondent à 
```suggestion
for (const [answerKey, answerValue] of Object.entries(answers)) {
```

Reply:
après concertation, on a pris la décision de laisser comme ça, considérant qu'une boucle for classique était plus lisible qu'une boucle sur l'`Object.entries`